hedera-node/hedera-token-service/src/test/java/com/hedera/node/app/service/token/AliasUtilsTest.java

```diff
@@ -0,0 +1,265 @@
+        assertTrue(AliasUtils.isSerializedProtoKey(ecdsaKeyAlias));
+    }
+
+    @Test
+    void isSerializedProtoKeyReturnsFalseForInvalidEcdsaKey() {
+        var invalidEcdsaKeyAlias = Bytes.wrap(HexFormat.of().parseHex(INVALID_ECDSA_KEY));
+        assertFalse(AliasUtils.isSerializedProtoKey(invalidEcdsaKeyAlias));
+    }
+
+    @Test
+    void isSerializedProtoKeyReturnsFalseForEvmAddress() {
+        var evmAddress = Bytes.wrap(HexFormat.of().parseHex(SAMPLE_EVM_ADDRESS.substring(2)));
+        assertFalse(AliasUtils.isSerializedProtoKey(evmAddress));
+    }
+
+    @Test
+    void extractShardFromAddressAliasReturnsCorrectValue() {
+        var alias = Bytes.wrap(HexFormat.of().parseHex(TEST_NON_ZERO_SHARD_ALIAS));
+        assertEquals(1, AliasUtils.extractShardFromAddressAlias(alias));
+    }
+
+    @Test
+    void extractShardFromAddressAliasReturnsZeroForZeroShard() {
+        var alias = Bytes.wrap(HexFormat.of().parseHex(TEST_ENTITY_NUM_ALIAS));
+        assertEquals(0, AliasUtils.extractShardFromAddressAlias(alias));
+    }
+
+    @Test
+    void extractRealmFromAddressAliasReturnsZeroForZeroRealm() {
+        var alias = Bytes.wrap(HexFormat.of().parseHex(TEST_ENTITY_NUM_ALIAS));
+        assertEquals(0L, AliasUtils.extractRealmFromAddressAlias(alias));
+    }
+
+    @Test
+    void extractIdFromAddressAliasReturnsCorrectValue() {
+        var alias = Bytes.wrap(HexFormat.of().parseHex(TEST_ENTITY_NUM_ALIAS));
+        assertEquals(1234L, AliasUtils.extractIdFromAddressAlias(alias));
+    }
+
+    @Test
+    void extractIdFromAddressAliasReturnsCorrectValueForNonZeroShardRealm() {
+        var alias = Bytes.wrap(HexFormat.of().parseHex(TEST_NON_ZERO_SHARD_ALIAS));
+        assertEquals(1234L, AliasUtils.extractIdFromAddressAlias(alias));
+    }
+
+    @Test
+    void isAliasReturnsTrueForAccountIdWithAlias() {
+        var alias = Bytes.wrap(HexFormat.of().parseHex(SAMPLE_EVM_ADDRESS.substring(2)));
+        var accountId = AccountID.newBuilder().alias(alias).build();
+        assertTrue(AliasUtils.isAlias(accountId));
+    }
+
+    @Test
+    void isAliasReturnsFalseForAccountIdWithAccountNum() {
+        var accountId = AccountID.newBuilder().accountNum(1234L).build();
+        assertFalse(AliasUtils.isAlias(accountId));
+    }
+
+    @Test
+    void isAliasReturnsTrueForAccountIdWithEvmAddressAlias() {
+        var evmAddress = Bytes.wrap(HexFormat.of().parseHex(SAMPLE_EVM_ADDRESS.substring(2)));
+        var accountId = AccountID.newBuilder().alias(evmAddress).build();
+        assertTrue(AliasUtils.isAlias(accountId));
+    }
+
+    @Test
+    void asKeyFromAliasReturnsKeyForValidEcdsaKey() {
+        var ecdsaKeyAlias = Bytes.wrap(HexFormat.of().parseHex(SAMPLE_ECDSA_PUBLIC_KEY));
+        var key = AliasUtils.asKeyFromAlias(ecdsaKeyAlias);
+        assertNotNull(key);
+        assertTrue(key.hasEcdsaSecp256k1());
+        assertEquals(33, key.ecdsaSecp256k1OrThrow().length());
```

Comment:
Why not calculate what the key should be here and assert equality, instead of just checking the length?

```suggestion
        assertEquals(ACTUAL_ECDSA_KEY, key.ecdsaSecp256k1OrThrow().length());
```

Also the same in `asKeyFromAliasPreCheckReturnsKeyForValidEcdsaKey()`

Reply:
Fixed 👍